src/utils/getFenceAndImportDiffsFromGit.ts

```diff
@@ -0,0 +1,249 @@
+import * as Git from 'nodegit';
+import { loadConfigFromString } from './loadConfig';
+import Config from '../types/config/Config';
+import normalizePath from './normalizePath';
+import NormalizedPath from '../types/NormalizedPath';
+import { reportWarning } from '../core/result';
+import { getScriptFileExtensions } from './getScriptFileExtensions';
+import { FenceDiff, getFenceDiff } from './getFenceDiff';
+import { getTsImportSetFromSourceString } from './getTsImportSetFromSourceString';
+
+/**
+ * Creates an empty fence given a path.
+ *
+ * Used when diffing fences against a fence that did not or no longer exists.
+ */
+function emptyFence(path: NormalizedPath): Config {
+    return {
+        tags: [],
+        imports: null,
+        exports: null,
+        dependencies: null,
+        path: path,
+    };
+}
+
+async function resolveToCommit(

```

Comment:
Can you break up these helper functions into separate files?  It might make sense to add a dir for all the partial diffing logic instead of dropping it all into `utils`.